### PR TITLE
Declare DB images defaults in parent POM

### PR DESCRIPTION
**Pull request by tsegismont**
Suggested as a follow-up PR in https://github.com/quarkusio/quarkus/pull/22810#discussion_r783084932

### CODE REVIEW

File: build-parent/pom.xml [hunk 1]
```diff
@@ -83,6 +83,13 @@
         <opensearch.image>opensearchproject/opensearch:${opensearch-server.version}</opensearch.image>
         <opensearch.protocol>http</opensearch.protocol>
 
+        <!-- Database images for JDBC/Reactive/Hibernate tests -->
+        <postgres.image>postgres:14.1</postgres.image>
+        <mariadb.image>mariadb:10.4</mariadb.image>
+        <db2.image>ibmcom/db2:11.5.5.0</db2.image>
+        <mssqldb.image>mcr.microsoft.com/mssql/server:2019-CU8-ubuntu-16.04</mssqldb.image>
```
Comment (gsmet) [inline build-parent/pom.xml:90]: Why `mssqldb` instead of `mssql`?

File: build-parent/pom.xml [hunk 2]
```diff
@@ -83,6 +83,13 @@
         <opensearch.image>opensearchproject/opensearch:${opensearch-server.version}</opensearch.image>
         <opensearch.protocol>http</opensearch.protocol>
 
+        <!-- Database images for JDBC/Reactive/Hibernate tests -->
+        <postgres.image>postgres:14.1</postgres.image>
+        <mariadb.image>mariadb:10.4</mariadb.image>
```
Comment (gsmet) [inline build-parent/pom.xml:88]: We had some issues with Podman default config requiring a registry to be defined so maybe it's a good opportunity to change the Docker Hub ones to `docker.io/...`.

**Review by gsmet — COMMENTED** [review]
Thanks a lot for this. Very nice cleanup. I added some questions/comments.

File: build-parent/pom.xml (hunk 1 above)
Comment (tsegismont) [reply build-parent/pom.xml:90]: I picked the name used originally in `integration-tests/jpa-mssql/pom.xml`.

Comment (gsmet) [reply build-parent/pom.xml]: OK, I adjusted it to be more consistent.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks @tsegismont , nice cleanup.

/cc @Sanne FYI when you will want to update them. Note that the Dev Services images are NOT yet handled this way so they also need an update.

**Comment by Sanne** [comment]
Very nice, thanks @tsegismont

**Pull request merged**